src/Model/ISO2ROI.py

```diff
@@ -0,0 +1,292 @@
+        self.isodose_levels = {}
+
+    def start_conversion(self, interrupt_flag, progress_callback):
+        """
+        Goes the the steps of the iso2roi conversion.
+        :param interrupt_flag: interrupt flag to stop process
+        :param progress_callback: signal that receives the current
+                                  progress of the loading.
+        """
+        patient_dict_container = PatientDictContainer()
+
+        progress_callback.emit(("Validating Datasets", 0))
+        val = ImageLoading.is_dataset_dicom_rt(patient_dict_container.dataset)
+
+        # Stop loading
+        if interrupt_flag.is_set():
+            print("Stopped ISO2ROI")
+            return False
+
+        if not val:
+            # Check if RT struct file is missing. If yes, create one and
+            # add its data to the patient dict container
+            if not patient_dict_container.get("file_rtss"):
+                # Stop loading
+                if interrupt_flag.is_set():
+                    print("Stopped ISO2ROI")
+                    return False
+
+                # Get common directory
+                file_path = patient_dict_container.filepaths.values()
+                file_path = Path(os.path.commonpath(file_path))
+
+                # Get new RT Struct file path
+                file_path = str(file_path.joinpath("rtss.dcm"))
+
+                # Create RT Struct file
+                progress_callback.emit(("Generating RT Structure Set", 30))
+                ds = ROI.create_initial_rtss_from_ct(
+                    patient_dict_container.dataset[0], file_path)
+                ds.save_as(file_path)
+
+                # Add RT Struct file path to patient dict container
+                patient_dict_container.filepaths['rtss'] = file_path
+                filepaths = patient_dict_container.filepaths
+
+                # Add RT Struct dataset to patient dict container
+                patient_dict_container.dataset['rtss'] = ds
+                dataset = patient_dict_container.dataset
+
+                # Set some patient dict container attributes
+                patient_dict_container.set("file_rtss", filepaths['rtss'])
+                patient_dict_container.set("dataset_rtss", dataset['rtss'])
+
+                dicom_tree_rtss = DicomTree(filepaths['rtss'])
+                patient_dict_container.set("dict_dicom_tree_rtss",
+                                           dicom_tree_rtss.dict)
+
+                dict_pixluts = ImageLoading.get_pixluts(
+                    patient_dict_container.dataset)
+                patient_dict_container.set("pixluts", dict_pixluts)
+
+                rois = ImageLoading.get_roi_info(ds)
+                patient_dict_container.set("rois", rois)
+
+                patient_dict_container.set("selected_rois", [])
+                patient_dict_container.set("dict_polygons_axial", {})
+
+        # Get isodose levels to turn into ROIs
+        self.get_iso_levels()
```

Comment:
a get function suggests it would return something, and `calculate_boundaries()` requires the latest iso_levels list, it's better to do something like:
```
iso_levels = self.get_iso_levels()
boundaries = self.calculate_boundaries(iso_levels)
```
and remove `self.isodose_levels` as well.

Let `get_iso_levels()` function return the `iso_levels` list instead of saving it to the class will help to avoid synchronization issues for the future, and put `iso_levels` as a parameter for `calculate_boundaries()` makes this function easier to reuse i think :D

Reply:
That is a good idea, however I'm not sure if there would be another use-case in the future for reading the `isodoseRoi.csv` file outside of calculating isodose ROIs and populating the table in the options menu. Despite this, I will implement these changes.

Reply:
Change has been made and added to the PR.